rfcs/0024-Non-consensus-node-protocol.md

```diff
@@ -0,0 +1,219 @@
+## Summary
+
+This RFC proposes a design for a protocol that allows users in low-capacity environments such as browsers, smartphones, and some desktops to verify the state of the network and perform operations such as retrieving account balance, sending payments etc with the same (similar) level of assurance as that of a full-node.
+
+The node that follows this protocol, is called a non-consensus node and can verify the state of the network without having to store the entire state. These nodes do not produce blocks but track enough information required to verify the state of the network. They rely on block producers and archive nodes for all other data and can easily validate them. The aim is to also reduce bandwidth usage by requiring only a subset of information from the network.
+
+## Requirements
+
+Coda daemon should by default run the non-consensus node and be able to switch to being consensus nodes
+
+A non-consensus node should perform the following operations:
+
+    1. Verify the state of the network
+    1. Get the state of an account (balance, nonce, receipt-chain hash)
+    2. Send payments
+    3. Check status of the payment made:
+        a) Seen (added to the pool?)
+        b) Processed
+        c) Finalized
+    4. View payments sent and received 
+        1. by querying an archive node maybe? (current wallet feature)
+    5. Delegate an account's stake to a block producer
+
+## Detailed design
+
+Coda daemon can currently run the following nodes that are able to fully verify the state and store the entire state:
+    1. Block producer
+    2. Full node (does not produce blocks) (can be a seed node, a snark worker, an archive node etc)
+
+A non-consensus (call it NC?) node is a full node but with just enough state to perform the above listed requirements. Let's call this a block header (similar to the what light clients require in other blockchains)? Block headers are gossiped to the non-consensus nodes by block producers, full nodes, and other non-consesus nodes. With these block headers an NC node can:
```

Comment:
Acronyms :(

Reply:
It’s okay to leave it here, but let’s not use nc acronym in code or conversation

Reply:
`noncon` is my current fave